packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanBar.tsx

```diff
@@ -81,12 +73,22 @@ function SpanBar(props: TInnerProps) {
     return toPercent(Math.round(posPercent * 500) / 500);
   });
 
+  const [label, setLabel] = useState(shortLabel);
+
+  const handleMouseOut = () => {
+    setLabel(shortLabel);
+  };
+
+  const handleMouseOver = () => {
+    setLabel(longLabel);
+  };
+
   return (
     <div
       className="SpanBar--wrapper"
       onClick={onClick}
-      onMouseOut={setShortLabel}
-      onMouseOver={setLongLabel}
+      onMouseOut={handleMouseOut}
+      onMouseOver={handleMouseOver}
```

Comment:
nit: can you keep the same names setShortLabel/setLongLabel? In fact, you're breaking the separation of concerns somewhat, `handleMouseOut` has nothing to do with mouse out in its function, it's just setting the label, and where it's _used_ does not need to be reflected in its name.

Reply:
Updated in latest push.